plan/physical_plans.go

```diff
@@ -103,17 +102,19 @@ func (p *physicalTableSource) addLimit(l *Limit) {
 	}
 }
 
-func (p *physicalTableSource) addTopN(prop *requiredProperty) {
+func (p *physicalTableSource) addTopN(prop *requiredProperty) bool {
 	if p.client == nil || !p.client.SupportRequestType(kv.ReqTypeSelect, kv.ReqSubTypeTopN) {
-		return
+		return false
 	}
-	if prop.limit != nil {
-		count := int64(prop.limit.Count + prop.limit.Offset)
-		p.LimitCount = &count
+	if prop.limit == nil || len(prop.props) == 0 {
```

Comment:
"prop.limit == nil" may not to be judged, it has been promised before invoking this function.